### ISSUE TITLE
new achievement

### ISSUE THREAD

**Issue by Omtrw** [fnk93/Summoners-War-Bot#15]
Is that possible to add the new achievement,? per example when completing Necro B7, B10 or ToA the achievement are not unlocked

**Comment by fnk93** [comment]
Yes it is. And that's what i'm doing currently: figuring out when exactly what achievement related calls are triggered.

**Comment by Omtrw** [comment]
Oh well, good luck for that and ty.

**Comment by fnk93** [comment]
You can already take a look into "achievements.json" where all achievements are listed. Right now their title is only available in korean.

**Comment by fnk93** [comment]
"Error logging reward_info" and "Error logging unit_list" don't necessarily mean something went wrong. I just haven't had the time to catch all special cases of requests returning an item called unit_list or reward_info. They vary from the whole list to just specific elements, so don't worry about that. If you could tell me after which requests you get this info, I can exactly tell you why that's printed out.

GetRTPvPInfo_v3 is failing to send data right now, because RTA season finished. But that's nothing to worry about either, as the official game app also sends this request and gets the same response.

**Comment by Omtrw** [comment]
Oh well, its working fine mb was I just not inoff patient lmao

**Comment by Omtrw** [comment]
@fnk93 Tomorrow is free energy, if I greed and put a 0s time on DB10, you think I'll get ban ? lmao

**Comment by fnk93** [comment]
Yes I Think so.

**Comment by Omtrw** [comment]
@fnk93 hm idk, I have already a sword of damocles above of my head, my best time is already 0s due to Mila's bot which have no timer.
Did you put a rune limit, can we get higher than 600 or not?

**Comment by Omtrw** [comment]
@fnk93 I was wondering, is that possible to farm SD?

**Comment by fnk93** [comment]
@Omtrw Rune limit is hardcoded by the server. You should get an error when trying to start a dungeon run with full rune inventory.
It is possible and I'm implementing a "do_instance" function right now. Problem with that is, that you will need to have a friend to get the sd you want, in its current state.
I try to come up with a mapping which lets you pick a unit for example which has a sd, and when the respective day comes, the bot will farm the essence dungeon until it drops a sd and and so on.
Right now there is no possibility to watch chat about other people opening the sd's you're looking for

**Comment by Omtrw** [comment]
@fnk93 kk thanks, was asking for rune because when summoning the limit can be outpassed with bot

**Comment by fnk93** [comment]
@Omtrw Okay, didnt' know that. I think I've implemented some checks on this in the mybot.py to not exceed the monster box/ monster storage limit.

**Comment by FR96** [comment]
@fnk93 might as well ask it here.
 Apparently there's an error when trying to do faimon volcano normal last stage with a new account. Everything worked well until that level, however, when doing that stage, the program exits with some error code. 
From what I can tell, either battleScenarioStart or battleScenarioEnd return something that's not correct, and when the program does parseBattleResult(battle_end) the log tries to format something that's not correct and you get the TypeError. 
(Some other weird stuff happend too if you're interested i'll mention it) 
Finally, I did that stage manually but got no reward for beating it (usually it's a L&D Scroll). After closing SW, I ran again the bot, and this time it kept doing the scenarios without issues (up until now at least)

**Comment by fnk93** [comment]
@FR96 I will need into the "official" sent data, as I logged every scenario with a fresh account. I think for getting the L&D Scroll there's an "event" that gets triggered which gives you the reward when finishing the first time.

**Comment by fnk93** [comment]
@FR96 Indeed there is an event triggering the sending of the L&D at the end of faimon:

UpdateEventStatus(567)
UpdateEventStatus(25)
UpdateEventStatus(26) -> This one gives the L&D scroll
UpdateEventStatus(27)

**Comment by laoli4121** [comment]
I find that every time while I am trying to update achievements 205 and 206 which refers to Giant B7 and B1, I'll be banned.
Still trying to figure out why server can filter my request this time with bot. Exactly step as below:
self.doDungeonAndSellRune(8001, 1)
		self.UpdateAchievement(self.makeList(203,1,1))
		self.doDungeonAndSellRune(8001, 2)
		self.doDungeonAndSellRune(8001, 3)
		self.doDungeonAndSellRune(8001, 4)
		self.UpdateAchievement(self.makeList(204, 1, 1))
		self.doDungeonAndSellRune(8001, 5)
		self.doDungeonAndSellRune(8001, 6)
		self.doDungeonAndSellRune(8001, 7, 1)
		self.UpdateAchievement(self.makeMultiList(2,[49,1,1,205,1,1]))

Have any idea to overcome this problem?

**Comment by fnk93** [comment]
Did you use fresh accounts? I've heard that they banned accounts with a low lvl completing gb7 for example. Did you already update the battledungeonstart API call to include the attribute "has_leader"?

**Comment by laoli4121** [comment]
No, API without leader, but I don't think it's a problem.
I complete that achievement at about lvl 9 or 10, maybe you right, too low lvl.

Will do more experiment to figure out.

**Comment by fnk93** [comment]
Should propably be no problem, you're right.
"leader_exist" is actually only used for BattleDungeonResult.

I guess you did UpdateEventStatus(1501) in your login-routine (or new-account routine)?
I will also try to figure out, what needs to be done to correctly update those achievements.
